test/cpp/jit/test_inline_cs_serialization.cpp

```diff
@@ -0,0 +1,147 @@
+    auto rhs_sr = std::get<1>(vec1[i]);
+    auto lhs_sr = std::get<1>(vec2[i]);
+    auto rhs_module = std::get<2>(vec1[i]);
+    auto lhs_module = std::get<2>(vec2[i]);
+    if (!((rhs_module.has_value() == lhs_module.has_value()) &&
+          (rhs_module.has_value() &&
+           (rhs_module.value().class_type()->name().value() ==
+            lhs_module.value().class_type()->name().value()) &&
+           (rhs_module.value().instance_name() ==
+            lhs_module.value().instance_name())) &&
+          (rhs_sr == lhs_sr))) {
+      return false;
+    }
+  }
+  return true;
+}
+
+TEST(InlineCSSerializaitionTest, TwoSubmodules) {
+  std::shared_ptr<CompilationUnit> cu = std::make_shared<CompilationUnit>();
+  Module a("A", cu);
+  a.define(R"JIT(
+    def forward(self, x):
+      return x + 1
+  )JIT");
+  Module b("B", cu);
+  b.define(R"JIT(
+    def forward(self, x):
+      return x + 2
+  )JIT");
+  Module c("C", cu);
+  c.register_module("A0", a);
+  c.register_module("B0", b);
+  c.define(R"JIT(
+    def forward(self, x):
+      return self.A0.forward(x) + self.B0.forward(x)
+  )JIT");
+
+  BackendDebugHandleManager debug_handle_manager;
+  auto graph = c.get_method("forward").graph();
+  Inline(*graph);
+  std::stack<Block*> blocks_to_visit;
+
+  // maps from source range to debug handle
+  SourceRangeTagMap source_range_tags;
+  // Maps from debug handle to source range
+  ska::flat_hash_map<int64_t, SourceRange> source_range_map;
+  int64_t source_range_tag{0};
+
+  blocks_to_visit.push(graph->block());
+  while (!blocks_to_visit.empty()) {
+    Block* b = blocks_to_visit.top();
+    blocks_to_visit.pop();
+    for (Node* n : b->nodes()) {
+      int64_t debug_handle{-1};
+      source_range_tags[n->sourceRange()] = source_range_tag;
+      source_range_map[source_range_tag] = n->sourceRange();
+      source_range_tag++;
+      if (n->callstack().has_value()) {
+        debug_handle_manager.getNextDebugHandleForInlinedCallStackPtr(
+            n->sourceRange(), n->callstack().value());
+        if (n->callstack().value()) {
+          for (const auto& e : n->callstack().value()->vec()) {
+            auto sr = std::get<1>(e);
+            source_range_tags[sr] = source_range_tag;
+            source_range_map[source_range_tag] = sr;
+            source_range_tag++;
+          }
+        }
+      } else {
+        debug_handle_manager.getNextDebugHandleForInlinedCallStackPtr(
+            n->sourceRange(), c10::intrusive_ptr<InlinedCallStack>());
+      }
+      for (Block* subblock : n->blocks()) {
+        blocks_to_visit.push(subblock);
+      }
+    }
+  }
+  auto debug_handle_cs_ptr_map = debug_handle_manager.getCallStackPtrMap();
+  InlinedCallStackPickler inlined_cs_pickler;
+  auto cs_data =
+      inlined_cs_pickler.pickle(debug_handle_cs_ptr_map, source_range_tags);
+  at::DataPtr data_ptr(cs_data.data(), DeviceType::CPU);
+  InlinedCallStackUnpickler unpickler;
+  // We wont really find any types
```

Comment:
This is actually inconsistent with the code. Earlier I had
```
...
  Module c("C", cu);
  c.register_module("A0", a);
  c.register_module("B0", b);
  c.define(R"JIT(
    def forward(self, x):
      return self.A0.forward(x) + self.B0.forward(x)
  )JIT");
...
  auto deserialized_cs_map = unpickler.unpickle(
      std::move(data_ptr), cs_data.size(), source_range_map, c._ivalue()->type->cu);
```
c._ivalue()->type->cu, part was getting cu of only c module and I had thought it would have types of all the modules which are part of c.
Now I create all the module within same compilation unit so we do actually find types.